cores/esp8266/abi.cpp

```diff
@@ -24,57 +24,67 @@
 using __cxxabiv1::__guard;
 
 // Debugging helper, last allocation which returned NULL
-extern void *umm_last_fail_alloc_addr;
-extern int umm_last_fail_alloc_size;
+extern "C" void *_heap_abi_malloc(size_t size, bool unhandled, const void* const caller);
 
 extern "C" void __cxa_pure_virtual(void) __attribute__ ((__noreturn__));
 extern "C" void __cxa_deleted_virtual(void) __attribute__ ((__noreturn__));
 
+#if defined(__cpp_exceptions) && (defined(DEBUG_ESP_OOM) || defined(DEBUG_ESP_PORT) || defined(DEBUG_ESP_WITHINISR))
```

Comment:
Shouldn't 'new\_handler' be independent of debugging?
ref new_op.cc and new_opnt.cc, nothrow variant just wraps the other in try catch

Reply:
> Shouldn't 'new_handler' be independent of debugging?

It has been a while since I did this, I think the increased build size for non-debug builds was a consideration.
It would add 136 bytes of IROM to the non-debug build so that it always builds with our revised `new_handler`.

Reply:
> Plus there are more missing overloads from -std=c++17

The current master does not support "new" align operations. Compiler reports:
 ` /workdir/repo/gcc-gnu/libstdc++-v3/libsupc++/new_opa.cc:86: undefined reference to `memalign'`

No hit on a quick issue check for `memalign`.

The library umm_malloc needs an enhancement to support an allocate-aligned option. Also, things will get complicated when incorporating the poison check feature.

Reply:
I've worked through adding an alignment option to the `umm_malloc` library. In the process, I have addressed an unreported alignment issue. `umm_malloc` allocated data addresses always landed on an 8-byte aligned - 4-byte. 
Now I need to finish working through all the build options in `abi.cpp` and `heap.cpp`.

Reply:
Can this be a config error? GCC / newlib / hal / etc.

Reading cppreference and some of the internal GCC stuff from '-faligned-new=X', 'Fundamental alignment' and max_align_t seem to come from 'long double' and 'long long' which are afaik 4-byte ones.

Currently, it does indeed show 8-byte
```
constexpr auto a = alignof(long long); // 8
constexpr auto b = alignof(long double); // 8
```

Reply:
> (_edit: comment thread from above_)
> 
> Can this be a config error? GCC / newlib / hal / etc.
> 
> Reading cppreference and some of the internal GCC stuff from '-faligned-new=X', 'Fundamental alignment' and max_align_t seem to come from 'long double' and 'long long' which are afaik 4-byte ones.
> 
> Currently, it does indeed show 8-byte
> 
> ```
> constexpr auto a = alignof(long long); // 8
> constexpr auto b = alignof(long double); // 8
> ```

...which is sort-of expected, as run-of-the-mill xtensa isa pdf does specify alignment req for long double and long long as 8. compiler here works according to spec, stack and heap alignment would stay consistent vs. current approach

no mention / explicit alignment things mentioned in the machine config, besides the `BIGGEST_ALIGNMENT` of 128 (bits)
https://github.com/gcc-mirror/gcc/tree/master/gcc/config/xtensa `/ALIGN/`

esp32 variants adhere to a similar standard. however, esp8266 rtos does use 4-byte heap alignment
(which I would assume works just fine for the most part)